lib/iris/common/resolve.py

```diff
@@ -57,12 +64,260 @@
+        self.lhs_cube_dim_coverage = None  # set in _metadata_coverage
+        #: Analysis of aux and scalar coordinates spanning the ``lhs`` :class:`~iris.cube.Cube`.
+        self.lhs_cube_aux_coverage = None  # set in _metadata_coverage
+        #: Analysis of dim coordinates spanning the ``rhs`` :class:`~iris.cube.Cube`.
+        self.rhs_cube_dim_coverage = None  # set in _metadata_coverage
+        #: Analysis of aux and scalar coordinates spanning the ``rhs`` :class:`~iris.cube.Cube`.
+        self.rhs_cube_aux_coverage = None  # set in _metadata_coverage
+
+        #: Map **common** metadata from the ``rhs`` :class:`~iris.cube.Cube` to
+        #: the ``lhs`` :class:`~iris.cube.Cube` if ``lhs-rank`` >= ``rhs-rank``,
+        #: otherwise map **common** metadata from the ``lhs`` :class:`~iris.cube.Cube`
+        #: to the ``rhs`` :class:`~iris.cube.Cube`.
+        self.map_rhs_to_lhs = None  # set in __call__
+
+        #: Mapping of the dimensions between **common** metadata for the :class:`~iris.cube.Cube`
+        #: operands, where the direction of the mapping is governed by
+        #: :attr:`~iris.common.resolve.Resolve.map_rhs_to_lhs`.
+        self.mapping = None  # set in _metadata_mapping
+
+        #: Cache containing a list of dim, aux and scalar coordinates prepared
+        #: and ready for creating and attaching to the resultant resolved
+        #: :class:`~iris.cube.Cube`.
+        self.prepared_category = None  # set in _metadata_prepare
+
+        #: Cache containing a list of aux factories prepared and ready for
+        #: creating and attaching to the resultant resolved
+        #: :class:`~iris.cube.Cube`.
+        self.prepared_factories = None  # set in _metadata_prepare
+
+        # The shape of the resultant resolved cube.
+        self._broadcast_shape = None  # set in _as_compatible_cubes
+
         if lhs is not None or rhs is not None:
+            # Attempt to resolve the cube operands.
             self(lhs, rhs)
 
     def __call__(self, lhs, rhs):
-        self._init(lhs, rhs)
+        from iris.cube import Cube
+
+        emsg = (
+            "{cls} requires {arg!r} argument to be a 'Cube', got {actual!r}."
+        )
+        clsname = self.__class__.__name__
+
+        if not isinstance(lhs, Cube):
+            raise TypeError(
+                emsg.format(cls=clsname, arg="LHS", actual=type(lhs))
+            )
+
+        if not isinstance(rhs, Cube):
+            raise TypeError(
+                emsg.format(cls=clsname, arg="RHS", actual=type(rhs))
+            )
+
+        # Initialise the operand state.
+        self.lhs_cube = lhs
+        self.rhs_cube = rhs
+
+        # Determine the initial direction to map operands.
+        # This may flip for operands with equal rank, particularly after
+        # later analysis informs the decision.
+        if self.lhs_cube.ndim >= self.rhs_cube.ndim:
+            self.map_rhs_to_lhs = True
+        else:
+            self.map_rhs_to_lhs = False
```

Comment:
Decided to make a slight infrastructure change to support the documentation of `Resolve` attribute members.

This involves pretty much taking the contents of `_init` and relocating it to `__init__` so that `sphinx` captures the doc-strings (`#:`) of `Resolve` attribute members (this **must** happen in `__init__`, otherwise `sphinx` won't document them), and ensuring any other necessary logic is placed here in `__call__`.